### PR TITLE
Blog app - creating a data model

### PR DESCRIPTION
**Pull request by banlon-jones**
- run the necessary migration files.
- Created table and column names based on the ERD diagram.
- Foreign keys included.

### CODE REVIEW

File: db/migrate/20220430103855_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreateUsers < ActiveRecord::Migration[6.1]
+  def change
+    create_table :users do |t|
+
+      t.timestamps
+    end
```
Comment (arslanbisharat) [inline db/migrate/20220430103855_create_users.rb:6]: -  [ ] There is no migration column added. Make sure you have proper migration created for user. 

image

File: db/migrate/20220430105035_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class CreatePosts < ActiveRecord::Migration[6.1]
+  def change
+    create_table :posts do |t|
+
+      t.timestamps
+    end
```
Comment (arslanbisharat) [inline db/migrate/20220430105035_create_posts.rb:6]: - [ ]  There is no migration column added. Make sure you have proper migration created for post.
image

File: db/migrate/20220430104830_create_comments.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CreateComments < ActiveRecord::Migration[6.1]
+  def change
+    create_table :comments do |t|
+
+      t.timestamps
+    end
```
Comment (arslanbisharat) [inline db/migrate/20220430104830_create_comments.rb:6]: - [ ]  There is no migration column added. Make sure you have proper migration created for comments.
image

File: db/migrate/20220430105409_create_likes.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class CreateLikes < ActiveRecord::Migration[6.1]
+  def change
+    create_table :likes do |t|
+
+      t.timestamps
+    end
```
Comment (arslanbisharat) [inline db/migrate/20220430105409_create_likes.rb:6]: - [ ]  There is no migration column added. Make sure you have proper migration created for likes

image

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] I can see you have created individual migration for each property but this makes the database hard to read and manage. Please try to follow the comments given below. 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- [ ]  Make sure you follow the correct gitflow. Currently, you have Github flow. 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: db/migrate/20220430103855_create_users.rb (hunk 1 above)
Comment (banlon-jones) [reply db/migrate/20220430103855_create_users.rb:6]: thanks for the feedback @arslanbisharat but I think you didn't go through my PR  carefully you would have the migrations
to `add_name_to_user`,`add_photo_to_user` and `add_post_counter_to_user`

File: db/migrate/20220430105035_create_posts.rb (hunk 2 above)
Comment (banlon-jones) [reply db/migrate/20220430105035_create_posts.rb:6]: thanks for the feedback @arslanbisharat but the migrations to `add_text_to_post`,`add_title_to_post` `add_comment_counter_to_post` and `add_comment_counter_to_post` are present in separate

File: db/migrate/20220430104830_create_comments.rb (hunk 3 above)
Comment (banlon-jones) [reply db/migrate/20220430104830_create_comments.rb:6]: same here

File: db/migrate/20220430105409_create_likes.rb (hunk 4 above)
Comment (banlon-jones) [reply db/migrate/20220430105409_create_likes.rb:6]: please check the schema.rb

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: db/migrate/20220430103855_create_users.rb (hunk 1 above)
Comment (arslanbisharat) [reply db/migrate/20220430103855_create_users.rb]: I can see you have created individual migration for each property but this makes the database hard to read and manage. Please try to follow the comments given below.

File: db/migrate/20220430105035_create_posts.rb (hunk 2 above)
Comment (arslanbisharat) [reply db/migrate/20220430105035_create_posts.rb]: I can see you have created individual migration for each property but this makes the database hard to read and manage. Please try to follow the comments given below.

File: db/migrate/20220430104830_create_comments.rb (hunk 3 above)
Comment (arslanbisharat) [reply db/migrate/20220430104830_create_comments.rb]: Same

File: db/migrate/20220430105409_create_likes.rb (hunk 4 above)
Comment (arslanbisharat) [reply db/migrate/20220430105409_create_likes.rb]: Please check the summary of my review.

**Pull request merged**